RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/tbar95.cpp,v
retrieving revision 1.125
diff -b -u -2 -r1.125 tbar95.cpp
--- tbar95.cpp	2004/04/20 22:17:21	1.125
+++ tbar95.cpp	2004/05/07 12:56:57
@@ -480,4 +480,17 @@
     const bool isVertical = HasFlag(wxTB_VERTICAL);
 
+    bool doRemap, doRemapBg, doTransparent;
+    if (wxSystemOptions::GetOptionInt(wxT("msw.remap")) == 2)
+    {
+        doRemapBg = doRemap = false;
+        doTransparent = true;       
+    }
+    else
+    {   doRemap = !wxSystemOptions::HasOption(wxT("msw.remap"))
+            || wxSystemOptions::GetOptionInt(wxT("msw.remap")) == 1;
+        doRemapBg = !doRemap;
+        doTransparent = false;
+    }
+
     // delete all old buttons, if any
     for ( size_t pos = 0; pos < m_nButtons; pos++ )
@@ -519,4 +532,7 @@
         wxBitmap bitmap(totalBitmapWidth, totalBitmapHeight);
         dcAllButtons.SelectObject(bitmap);
+        if (doTransparent)
+            dcAllButtons.SetBackground(*wxTRANSPARENT_BRUSH);
+        else
         dcAllButtons.SetBackground(*wxLIGHT_GREY_BRUSH);
         dcAllButtons.Clear();
@@ -543,5 +559,5 @@
 #endif // USE_BITMAP_MASKS/!USE_BITMAP_MASKS
 
-        if (wxSystemOptions::HasOption(wxT("msw.remap")) && wxSystemOptions::GetOptionInt(wxT("msw.remap")) == 0)
+        if (doRemapBg)
         {
 #if USE_BITMAP_MASKS
@@ -609,5 +625,5 @@
 #endif // USE_BITMAP_MASKS/!USE_BITMAP_MASKS
 
-        if (!wxSystemOptions::HasOption(wxT("msw.remap")) || wxSystemOptions::GetOptionInt(wxT("msw.remap")) == 1)
+        if (doRemap)
         {
             // Map to system colours

 	  	 
